File: hungry_person/hungry_person.py
from flask import Flask, request
import requests
app = Flask(__name__)

BAKER_ENDPOINT = 'http://baker:5000/v1/order'

@app.route('/v1/order', methods=['POST'])
def make_an_order():
    order = yell_the_order()
    return say_what_was_received(order.json())


def yell_the_order():
    return requests.post(BAKER_ENDPOINT, json=request.get_json())

def say_what_was_received(order):
    return "I received %s %s made by %s" % (order['quantity'],
                                            order['name'],
                                            order['madeBy'])

File: shop-floor/src/main/java/HungryPerson.java
import org.json.JSONObject;
import spark.Request;
import spark.Response;

import java.util.concurrent.BlockingQueue;

class HungryPerson {

    private final BlockingQueue<String> queue;
    private final ShopAssistant shopAssistant;

    HungryPerson(BlockingQueue<String> queue, ShopAssistant shopAssistant) {
        this.queue = queue;
        this.shopAssistant = shopAssistant;
    }

    String makeAnOrder(spark.Request req, Response res) throws Exception {
        return shopAssistant.iWant(req.body());
    }

    String showMeWhatYouGot(Request request, Response response) throws Exception {
        StringBuilder stringBuilder = new StringBuilder();

        while (!queue.isEmpty()) {
            stringBuilder.append(sayWhatWasReceived(queue.take()));
            stringBuilder.append("\n");
        }

        return stringBuilder.toString();
    }

    private String sayWhatWasReceived(String response) {
        JSONObject order = new JSONObject(response);
        return String.format("I received %s %s made by %s",
                order.get("quantity"),
                order.get("name"),
                order.get("madeBy"));
    }
}
